### ISSUE TITLE
Conversion tool from mimir helm chart 2.1.0 to 3.0.0

### ISSUE THREAD

**Issue by WilliamCassanova** [grafana/mimir#2646]
#### Is your feature request related to a problem? Please describe.

I have been attempting to get 3.0.0 working following the instructions written in https://github.com/grafana/mimir/blob/main/docs/sources/operators-guide/deploy-grafana-mimir/upgrade-helm-chart-2.1-to-3.0/_index.md but with limited success.

#### Describe the solution you'd like

A tool that could take my existing mimir 2.1.0 yaml files and convert them to something usable with 3.0.0.

#### Describe alternatives you've considered

Lots of debugging but the problem being that although some of the pods do come up they do not stay up long enough for me to extract what the corresponding generated config using something like this as an example:  `kubectl exec mimir-store-gateway-0 -- cat /etc/mimir/mimir.yaml`

Additional documentation on how to enable full debug log output for all pods.  I attempted:
```
server:
  log_level: debug
```
But wasn't about to glean any useful information about why my S3 endpoint was invalid.

```
  mimir:
    structuredConfig:
      blocks_storage:
        backend: s3
        tsdb:
          dir: /data/tsdb
        s3:
          access_key_id: ${AWS_ACCESS_KEY_ID}
          bucket_name: mimir-tsdb
          endpoint: s3.private.us-east.cloud-object-storage.appdomain.cloud
          secret_access_key: ${AWS_SECRET_ACCESS_KEY}
  global:
    extraEnvFrom:
      - secretRef:
          name: mimir-bucket-creds 
```

#### Additional context
This issue was originally noted in https://github.com/grafana/mimir/issues/2579 and I am about 98% certain my config is correct.

**Comment by WilliamCassanova** [comment]
I modified my config slightly to test and fully specified the region 
```
  mimir:
    structuredConfig:
      blocks_storage:
        backend: s3
        s3:
          access_key_id: ${AWS_ACCESS_KEY_ID}
          bucket_name: mimir-tsdb-south-auth
          endpoint: s3.private.us-south.cloud-object-storage.appdomain.cloud
          secret_access_key: ${AWS_SECRET_ACCESS_KEY}
          region: us-south
  global:
    extraEnvFrom:
      - secretRef:
          name: mimir-bucket-creds 
```

If it helps any the pods that are failing to start are:
```
kubectl get pods
NAME                                   READY   STATUS    RESTARTS      AGE
mimir-chunks-cache-0                   2/2     Running   0             13m
mimir-compactor-0                      0/1     Running   0             13m
mimir-distributor-744dfd46db-4glxg     1/1     Running   0             13m
mimir-distributor-744dfd46db-4pkdk     1/1     Running   0             13m
mimir-distributor-744dfd46db-9f9k9     1/1     Running   0             13m
mimir-distributor-744dfd46db-qcr7d     1/1     Running   0             13m
mimir-distributor-744dfd46db-vflqc     1/1     Running   0             13m
mimir-index-cache-0                    2/2     Running   0             13m
mimir-ingester-0                       0/1     Running   0             13m
mimir-ingester-1                       0/1     Running   0             13m
mimir-ingester-2                       0/1     Running   0             13m
mimir-ingester-3                       0/1     Running   0             13m
mimir-ingester-4                       0/1     Running   0             13m
mimir-ingester-5                       0/1     Running   0             13m
mimir-metadata-cache-0                 2/2     Running   0             13m
mimir-nginx-6dc68bb989-jch7m           1/1     Running   0             13m
mimir-querier-6f98fc7c55-2fcn5         0/1     Running   3 (12m ago)   13m
mimir-querier-6f98fc7c55-8rn9g         0/1     Running   3 (12m ago)   13m
mimir-querier-6f98fc7c55-f6dcw         0/1     Running   3 (12m ago)   13m
mimir-querier-6f98fc7c55-srx7w         0/1     Running   3 (12m ago)   13m
mimir-query-frontend-bb5f9f69b-phs94   0/1     Running   0             13m
mimir-store-gateway-0                  0/1     Running   2 (12m ago)   13m
```

The error in the logs:
```
Unable to successfully connect to configured object storage (will retry)" err="blocks storage: unable to successfully send a request to object storage: Get \"https://s3.private.us-south.cloud-object-storage.appdomain.cloud/mimir-tsdb-south-auth/sanity-check-at-startup\": context deadline exceeded"
```

**Comment by Thrinadh-Kumpatla** [comment]
Describe the node on which the pod is scheduled using `aws ec2 describe-instances --instance-ids  <instance_id>`, and check for 
`HttpPutResponseHopLimit` variable. This value should be greater than 2 in order to get response from s3 Bucket. You can check for reference [here](https://support.hashicorp.com/hc/en-us/articles/1500005343862-Required-Additional-Configuration-When-Using-IMDSv2).

**Comment by WilliamCassanova** [comment]
Thanks.  I will give that a try.

**Comment by kovaxur** [comment]
Hi,
We have the same problem with the S3 access, but we are migrating from Cortex to Mimir. With cortex everything was fine, can the HttpPutResponseHopLimit still be the problem? It's set to 1 currently.

thanks

**Comment by WilliamCassanova** [comment]
I did not get very far with testing the newest version....We pinned at `2.1.0` for the time-being.  I will be back to this after next week.  The one thing that I wish was available somewhere was something that could convert our existing Grafana dashboards from being a single cluster to something that is `mimir` aware.  We wrote some python scripting around it to modify the regular expressions but its a work in progress.
